### ISSUE TITLE
bug in the metrics api handing of missing resources (content return)

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#5622]
![screen shot 2016-07-12 at 12 32 29](https://cloud.githubusercontent.com/assets/68540/16764175/fb8f393e-482c-11e6-8de2-3e1485778f62.png)

when hitting a missing resources in the api, the user gets a 404 (what is the right behaviour), but it also gets back a hash full of content, what is call metadata. The expected behaviour of a 404 code would be to give back 404 plus empty response back as nothing was found to properly answer the request.

is there any missing benefit of having always included this metadata in the requests that I might be missing? specially for error situations? 

related https://github.com/elastic/logstash/pull/5621

**Comment by ph** [comment]
I think we should do the following:

- Return a 404 http code
- Return an tailored json when error happen.
- Explicitly show `status; 404` in the json

This give us more flexibility in the future and also align with other elastic product like ES.

By flexibility I meant human and computer can understand what happen.


As a reference this is the ES output.
```json
{  
   "error":{  
      "root_cause":[  
         {  
            "type":"index_not_found_exception",
            "reason":"no such index",
            "resource.type":"index_or_alias",
            "resource.id":"dddd",
            "index_uuid":"_na_",
            "index":"dddd"
         }
      ],
      "type":"index_not_found_exception",
      "reason":"no such index",
      "resource.type":"index_or_alias",
      "resource.id":"dddd",
      "index_uuid":"_na_",
      "index":"dddd"
   },
   "status":404
}
```

**Comment by purbon** [comment]
@ph like your idea is a good middle approach.

**Comment by ph** [comment]
cc @jordansissel

**Comment by ph** [comment]
fixed in #5622
